test/e2e/suite_test.go

```diff
@@ -1,4 +1,4 @@
-//go:build e2e
+//go:build e2e || e2e_healthchecks
```

Comment:
The e2e test is meant to run on different providers, since these health check tests are only working with the inmemory provider adding them to the normal e2e test suite seems pointless, you will just end up running the same test multiple times.